service/search/es.go

```diff
@@ -161,118 +174,205 @@ var esSetting = m{
 
 // NewESEngine Elasticsearch検索エンジンを生成します
 func NewESEngine(mm message.Manager, cm channel.Manager, repo repository.Repository, logger *zap.Logger, config ESEngineConfig) (Engine, error) {
-	// es接続
-	client, err := elastic.NewClient(elastic.SetURL(config.URL), elastic.SetSniff(false))
+	// esクライアント作成
+	client, err := elasticsearch.NewClient(elasticsearch.Config{
+		Addresses: config.URL,
+	})
 	if err != nil {
-		return nil, fmt.Errorf("failed to init search engine: %w", err)
+		return nil, fmt.Errorf("failed to init Elasticsearch: %w", err)
 	}
 
 	// esバージョン確認
-	version, err := client.ElasticsearchVersion(config.URL)
+	infoRes, err := client.Info()
+	defer infoRes.Body.Close()
 	if err != nil {
-		return nil, fmt.Errorf("failed to fetch es version: %w", err)
+		return nil, fmt.Errorf("failed to get Elasticsearch info: %w", err)
+	}
+	if infoRes.IsError() {
+		return nil, fmt.Errorf("failed to get Elasticsearch info: %s", infoRes.String())
+	}
+
+	var r map[string]interface{}
+	infoResBody, err := io.ReadAll(infoRes.Body)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read Elasticsearch info: %w", err)
+	}
+	err = json.Unmarshal(infoResBody, &r)
```

Comment:
New系関数だからパフォーマンスはあまり気にしなくていいんだけど、io.Readerは一回読み出すよりもそのまま処理した方が速いしメモリも食わないので好ましいです
このライブラリだとこの方がシンプルに書けそうだし
参考: https://pkg.go.dev/github.com/json-iterator/go@v1.1.12#Decoder
```suggestion
	err = json.NewDecoder(infoRes.Body).Decode(&r)
```